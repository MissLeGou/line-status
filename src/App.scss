.esub-page {
  width: 100%;
  .esub-page-header {
    justify-content: center;
  }

  .esub-table-row {
    column-count: 2;
    display: block;
  }

  @media screen and (max-width: 991px) {
    .esub-table-row {
      column-count: 1;
      display: block;
    }
  }
}
