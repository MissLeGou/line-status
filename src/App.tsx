import { Col, Container, Row } from "react-bootstrap";
import { EsubTable } from "./components/EsubTable";
import { useGetLineData } from "./hooks/useGetLineData";
import { getLineColor } from "./utils";
import "./App.scss";

function App() {
  const [linesData] = useGetLineData();

  return (
    <div className="App">
      <Container className="mt-5 mb-3 esub-page">
        <Row className="mb-3 esub-page-header">
          <h1>Line Status Board</h1>
        </Row>
        <Row className="esub-page-row">
          <Col sm={4} md={12} lg={12}>
            <Row className="esub-table-row">
              <Col>
                {Object.values(linesData).map(
                  (line: {
                    id: string;
                    name: string;
                    lineStatuses: [{ statusSeverityDescription: string }];
                  }) => (
                    <div key={line.id}>
                      <EsubTable
                        lineColor={getLineColor(line.id)}
                        lineName={line.name}
                        lineStatus={
                          line.lineStatuses[0].statusSeverityDescription
                        }
                      />
                    </div>
                  )
                )}
              </Col>
              <Col md={4} lg={4}></Col>
            </Row>
          </Col>
        </Row>
      </Container>
    </div>
  );
}

export default App;
