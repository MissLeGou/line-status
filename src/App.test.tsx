import { render, screen } from "@testing-library/react";
import ReactTestRenderer from "react-test-renderer";
import App from "./App";

describe("When app starts up", () => {
  it("renders correctly", () => {
    const renderer = ReactTestRenderer.create(<App />).toJSON();
    expect(renderer).toMatchSnapshot();
  });

  it("renders the App", async () => {
    render(<App />);

    await screen.findByText(/Line Status Board/);
  });

  it("fetches and displays line status data", async () => {
    render(<App />);

    const lineData = await screen.findByText("Bakerloo");
    expect(lineData).toBeInTheDocument();
  });
});
