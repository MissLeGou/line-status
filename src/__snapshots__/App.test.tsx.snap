// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When app starts up renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="mt-5 mb-3 esub-page container"
  >
    <div
      className="mb-3 esub-page-header row"
    >
      <h1>
        Line Status Board
      </h1>
    </div>
    <div
      className="esub-page-row row"
    >
      <div
        className="col-lg-12 col-md-12 col-sm-4"
      >
        <div
          className="esub-table-row row"
        >
          <div
            className="col"
          />
          <div
            className="col-lg-4 col-md-4"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
